pkg/test/e2e/default-app-catalog-applications-test/argocd/argocd_test.go

```diff
@@ -0,0 +1,180 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package argocd
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/go-logr/zapr"
+	"go.uber.org/zap"
+	apiv1 "k8c.io/kubermatic/v2/pkg/api/v1"
+	appskubermaticv1 "k8c.io/kubermatic/v2/pkg/apis/apps.kubermatic/v1"
+	kubermaticv1 "k8c.io/kubermatic/v2/pkg/apis/kubermatic/v1"
+	"k8c.io/kubermatic/v2/pkg/log"
+	"k8c.io/kubermatic/v2/pkg/test/e2e/jig"
+	"k8c.io/kubermatic/v2/pkg/test/e2e/utils"
+	"k8c.io/kubermatic/v2/pkg/util/wait"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/clientcmd"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	ctrlruntimelog "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var (
+	userconfig  string
+	credentials jig.AWSCredentials
+	logOptions  = utils.DefaultLogOptions
+)
+
+const (
+	projectName = "argocd-test-project"
+	argoCDNs    = "argocd"
+	argoCDName  = "argocd"
+)
+
+func init() {
+	flag.StringVar(&userconfig, "userconfig", "", "path to kubeconfig of usercluster")
+	credentials.AddFlags(flag.CommandLine)
+	jig.AddFlags(flag.CommandLine)
+	logOptions.AddFlags(flag.CommandLine)
+}
+
+func TestArgoCDClusters(t *testing.T) {
+	rawLog := log.NewFromOptions(logOptions)
+	logger := rawLog.Sugar()
+	ctx := context.Background()
+
+	if err := credentials.Parse(); err != nil {
+		t.Fatalf("Failed to get credentials: %v", err)
+	}
+
+	config, err := clientcmd.BuildConfigFromFlags("", os.Getenv("KUBECONFIG"))
+	if err != nil {
+		t.Fatalf("failed to build config: %v", err)
+	}
+
+	seedClient, err := ctrlruntimeclient.New(config, ctrlruntimeclient.Options{})
+	if err != nil {
+		t.Fatalf("failed to build ctrlruntime client: %v", err)
+	}
+
+	// set the logger used by sigs.k8s.io/controller-runtime
+	ctrlruntimelog.SetLogger(zapr.NewLogger(rawLog.WithOptions(zap.AddCallerSkip(1))))
+
+	client, cleanup, tLogger, err := createUserCluster(ctx, t, logger, seedClient)
+	if cleanup != nil {
+		defer cleanup()
+	}
+
+	if err != nil {
+		t.Fatalf("failed to create user cluster: %v", err)
+	}
+
+	testUserCluster(ctx, t, tLogger, client)
+
+}
+
+//gocyclo:ignore
+func testUserCluster(ctx context.Context, t *testing.T, log *zap.SugaredLogger, client ctrlruntimeclient.Client) {
+	log.Info("Running ArgoCD tests...")
+
+	log.Info("Checking for argocd ApplicationInstallation...")
+	err := wait.PollLog(ctx, log, 2*time.Second, 5*time.Minute, func(ctx context.Context) (error, error) {
+		app := &appskubermaticv1.ApplicationInstallation{}
+		if err := client.Get(context.Background(), types.NamespacedName{Namespace: argoCDNs, Name: argoCDName}, app); err != nil {
+			return fmt.Errorf("failed to get ArgoCD ApplicationInstallation in user cluster: %w", err), nil
+		}
+		if app.Status.ApplicationVersion == nil {
+			return fmt.Errorf("application not yet installed: %v", app.Status), nil
+		}
```

Comment:
This should _probably_ also check if the app is somewhat healthy (e.g. if its Pods are running). I'm not sure if the application-controller publishes such information to the `ApplicationInstallation` status, but if not, we should maybe try to look for the pods created by the app.